### PR TITLE
tabby-agent: init at 0.18.0

### PR DESCRIPTION
**Pull request by khaneliman**
Follow up to https://github.com/NixOS/nixpkgs/pull/350323 and predecessor for vim-tabby update https://github.com/NixOS/nixpkgs/pull/350532#issuecomment-2430070025

https://github.com/TabbyML/tabby/tree/main/clients/tabby-agent

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 351321`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tabby-agent</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tabby-agent</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tabby-agent</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tabby-agent</li>
  </ul>
</details>

File: pkgs/by-name/ta/tabby-agent/package.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  };
+
+  passthru.updateScript = nix-update-script {
+    extraArgs = [
+      "--version-regex"
+      "^v([0-9.]+)$"
+    ];
+  };
+
+  meta = with lib; {
```
Comment (GaetanLepage) [inline pkgs/by-name/ta/tabby-agent/package.nix:61]:
```suggestion
  meta = {
```
It is now discouraged to use the `with` pattern in nixpkgs.

File: pkgs/by-name/ta/tabby-agent/package.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    chmod +x $out/bin/tabby-agent
+
+    runHook postInstall
+  '';
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-WBhkrgLTTR5f8ZVmUfzMbFw/6jIGoLcUspHCsNpOxx4=";
+  };
+
```
Comment (GaetanLepage) [inline pkgs/by-name/ta/tabby-agent/package.nix:53]:
```suggestion
  };
  
  nativeInstallCheckInputs = [
    versionCheckHook
  ];
  versionCheckProgramArg = [ "--version" ];
  doInstallCheck = true;

```

File: pkgs/by-name/ta/tabby-agent/package.nix [hunk 3]
```diff
@@ -0,0 +1,69 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  nix-update-script,
+  nodejs,
+  pnpm,
+  wrapGAppsHook3,
+}:
+stdenv.mkDerivation (finalAttrs: {
```
Comment (GaetanLepage) [inline pkgs/by-name/ta/tabby-agent/package.nix:10]: Have you tried using `buildNodePackage` ?

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ta/tabby-agent/package.nix (hunk 2 above)
Comment (khaneliman) [reply pkgs/by-name/ta/tabby-agent/package.nix:53]: This binary doesn't return version information

File: pkgs/by-name/ta/tabby-agent/package.nix (hunk 3 above)
Comment (khaneliman) [reply pkgs/by-name/ta/tabby-agent/package.nix:10]: buildNodePackage doesn't work with pnpm lockfiles so we use mkDerivation.

Comment (GaetanLepage) [reply pkgs/by-name/ta/tabby-agent/package.nix:10]: Ok, very clear !

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
~~Maybe sneak in a `versionProgramHook` in there if possible.~~

Pardon my catastrophic memory

File: pkgs/by-name/ta/tabby-agent/package.nix (hunk 2 above)
Comment (GaetanLepage) [reply pkgs/by-name/ta/tabby-agent/package.nix:53]: Ok, good to know.

**Comment by GaetanLepage** [comment]
Can you try re-pushing to the branch ? The CI failure is a little bit weird.

**Comment by khaneliman** [comment]
> Can you try re-pushing to the branch ? The CI failure is a little bit weird.

Rebased and looked like CI failure went away.

File: pkgs/by-name/ta/tabby-agent/package.nix [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    extraArgs = [
+      "--version-regex"
+      "^v([0-9.]+)$"
+    ];
+  };
+
+  meta = {
+    homepage = "https://github.com/TabbyML/tabby";
+    changelog = "https://github.com/TabbyML/tabby/releases/tag/v${finalAttrs.version}";
+    description = "Language server used to communicate with Tabby server.";
```
Comment (sikmir) [inline pkgs/by-name/ta/tabby-agent/package.nix:64]:
```suggestion
    description = "Language server used to communicate with Tabby server";
```

**Review by GaetanLepage — APPROVED** [review]
LGTM !

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 351321`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tabby-agent</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tabby-agent</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tabby-agent</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tabby-agent</li>
  </ul>
</details>

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**